influxdb/src/main/scala/akka/stream/alpakka/influxdb/impl/InfluxDBFlowStage.scala

```diff
@@ -58,28 +61,70 @@ private[influxdb] class InfluxDBFlowStage[T, C](
   final class InfluxDBRecordLogic extends InfluxDBLogic {
 
     override protected def write(messages: immutable.Seq[InfluxDBWriteMessage[T, C]]): Unit =
-      messages.foreach {
-        case InfluxDBWriteMessage(point: Point, _) => {
-          influxDB.write(point)
+      messages
+        .filter {
+          case InfluxDBWriteMessage(point: Point, _, _, _) => {
+            true
+          }
+          case InfluxDBWriteMessage(others: AnyRef, _, _, _) => {
+            failStage(new RuntimeException(s"unexpected type Point or annotated with Measurement required"))
+            false
+          }
         }
-        case InfluxDBWriteMessage(others: AnyRef, _) =>
-          failStage(new RuntimeException(s"unexpected type Point or annotated with Measurement required"))
-      }
+        .groupBy(im => (im.databaseName, im.retentionPolicy))
+        .map(wm => toBatchPoints(wm._1._1, wm._1._2, wm._2))
+        .foreach(influxDB.write)
 
   }
 
   final class InfluxDBMapperRecordLogic extends InfluxDBLogic {
 
-    protected var influxDBMapper: InfluxDBMapper = _
-
-    override def preStart(): Unit =
-      influxDBMapper = new InfluxDBMapper(influxDB)
+    private val mapperHelper: InfluxDBResultMapperHelper = new InfluxDBResultMapperHelper
 
     override protected def write(messages: immutable.Seq[InfluxDBWriteMessage[T, C]]): Unit =
-      messages.foreach {
-        case InfluxDBWriteMessage(typeMetric: Any, _) =>
-          influxDBMapper.save(typeMetric)
+      messages
```

Comment:
One suggestion for readability around flows is keeping the flow declarations succinct. If we define methods elsewhere that handle the `groupBy` logic and the `map` logic, we can write this like:
```
messages
  .groupBy(groupByDbRp)
  .map(toBatchPoints)
  .foreach(influxDB.write)

def groupByDbRp(im: InfluxDBWriteMessage[T, C]) = ...
def toBatchPoints(wm: Thing) = ...
```
I think this helps communicate to users/readers the intent of the flow better.